core/src/main/java/org/elasticsearch/action/bulk/BulkItemResponse.java

```diff
@@ -90,9 +106,51 @@ public RestStatus getStatus() {
             return this.status;
         }
 
+        /**
+         * The actual cause of the failure.
+         */
         public Throwable getCause() {
             return cause;
         }
+
+        @Override
+        public void readFrom(StreamInput in) throws IOException {
+            index = in.readString();
+            type = in.readString();
+            id = in.readOptionalString();
+            cause = in.readThrowable();
+            status = ExceptionsHelper.status(cause);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(getIndex());
+            out.writeString(getType());
+            out.writeOptionalString(getId());
+            out.writeThrowable(getCause());
+        }
+
+        static final class Fields {
```

Comment:
can we just make these `XContentBuilderString` static a `static final String` in the enclosing class? I know we do this everywhere but it's really just confusing and not helpful.